main/vstring.c

```diff
@@ -156,13 +156,14 @@ extern void vStringStripNewline (vString *const string)
  */
 extern void vStringStripLeading (vString *const string)
 {
-	while (isspace ((int) string->buffer [0]) && string->length > 0)
+	size_t n = 0;
+
+	while (n < string->length && isspace ((int) string->buffer [n]))
+		n++;
+	if (n > 0)
 	{
-		size_t i;
-		for (i = 1  ;  i < string->length  ;  ++i)
-			string->buffer [i - 1] = string->buffer [i];
-		--string->length;
-		string->buffer [string->length] = '\0';
+		memmove (string->buffer, string->buffer + n, string->length - n);
+		vStringTruncate (string, string->length - n);
```

Comment:
My single line alternative sets the length and copies one more than it (since the zero term is already in the buffer, it just `memmove`s it), though I'm not sure I'm fond of tucking an assignment side effect into a function call argument like that :) It would be clearer to put it on two lines, but it's mostly pointless since in release build most/all of the overhead of the way in this PR goes away.